mautic/mautic
# Segments form submission filter

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              |  N/A
| Related user documentation PR URL      | N/A
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     |  N/A




#### Description:

Added 5 types of segment filters:
* form submission
* first conversion
* last conversion
* first conversion date
* last conversion date

With "Form Submission" a user can filter by form submission, a segment is populated with all the leads who submitted that form.

With "Last conversion" and "First conversion" a segment is populated with all the leads who submitted that form first (or last) .

With "First conversion date" and "Last conversion date" you can populate the segment filtering with the date of the first submission ( or the last submission).



#### Steps to test this PR:
1. Ensure you have multiple segments with multiple filters already before you check out this PR so you could test that existing segments still work.
2. Once you check out this branch, run the `bin/console mautic:segments:update` command and check that the command worked as before and no contacts were added or removed. The contact count for each existing segment should be the same
3. Edit a segment and check the functionality of the new filters





<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9694"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
